contracts/BirdSwap.sol

```diff
@@ -94,38 +95,36 @@ contract BirdSwap is IBirdSwap, UUPSUpgradeable, ReentrancyGuardUpgradeable, IER
     function setAskPrice(
         uint256 _tokenId,
         uint256 _askPrice
-    ) external nonReentrant {
+    ) external onlyTokenSeller(_tokenId) nonReentrant {
         Ask storage ask = askForMoonbird[_tokenId];
-
-        require(ask.seller == msg.sender || (isMoonbirdEscrowed(_tokenId) && moonbirdTransferredFromOwner[_tokenId] == msg.sender), "setAskPrice must be seller");
+        require(ask.seller == msg.sender, "setAskPrice must be seller");
+        require(_askPrice < ask.askPrice, "setAskPrice can only be used to lower the price");
 
         ask.askPrice = _askPrice;
 
-        emit AskPriceUpdated(_tokenId, ask);
+        emit AskPriceUpdated(_tokenId, ask.seller, ask.buyer, ask.askPrice, ask.royaltyFeeBps, ask.uid);
     }
 
-    // /// @notice Withdraws an Escrowed bird for a seller
-
-    // /// @dev note: cancelAsk is preferred for allowing a seller to withdrawing their bird
-    // function withdrawBird(uint256 _tokenId) external onlyTokenSeller(_tokenId) nonReentrant {
-    //     _withdrawBird(_tokenId);
-    // }
+    /// @notice Withdraws an Escrowed bird for a seller
+    /// @dev note: cancelAsk is preferred for allowing a seller to withdrawing their bird
+    // @param _tokenId the ID of the Moonbird token
+    function withdrawBird(uint256 _tokenId) external onlyTokenSeller(_tokenId) nonReentrant {
+        require(isMoonbirdEscrowed(_tokenId), "Bird not escrowed");
+        _withdrawBird(_tokenId);
+    }
 
     /// @notice Fills the ask for a given Moonbird, transferring the ETH/ERC-20 to the seller and Moonbird to the buyer
     /// @param _tokenId The ID of the Moonbird token
-    /// @param _fillAmount The amount to fill the ask
     function fillAsk(
-        uint256 _tokenId,
-        uint256 _fillAmount
+        uint256 _tokenId
     ) external payable nonReentrant {
         Ask storage ask = askForMoonbird[_tokenId];
 
         require(isMoonbirdEscrowed(_tokenId), "fillAsk The Moonbird associated with this ask must be escrowed within Birdswap before a purchase can be completed");
         require(ask.seller != address(0), "fillAsk must be active ask");
         require(ask.buyer == msg.sender, "must be buyer");
-        require(ask.askPrice == _fillAmount, "fillAsk _fillAmount must match ask amount");
 
-        require(msg.value >= ask.askPrice, "_handleIncomingTransfer msg value less than expected amount");
+        require(msg.value == ask.askPrice, "_handleIncomingTransfer msg value less than expected amount");
```

Comment:
suggest updating error message to reflect change in conditional "_handleIncomingTransfer msg value is not expected amount"